File: projects/order-creation/core/src/main/java/com/robinfood/core/mappers/ReplacementPortionMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.transactionrequestdto.ReplacementPortionDTO;
import com.robinfood.core.entities.ReplacementPortionEntity;

public final class ReplacementPortionMappers {

    private ReplacementPortionMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static ReplacementPortionEntity toReplacementPortionDTO(
            ReplacementPortionDTO replacementPortionDTO
    ) {
        return new ReplacementPortionEntity(
                replacementPortionDTO.getId(),
                replacementPortionDTO.getName()
        );
    }

}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/ReplacementPortionMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.transactionrequest.ReplacementPortionDTO
import com.robinfood.core.entities.transactionrequest.ReplacementPortionEntity

fun ReplacementPortionDTO.toReplacementPortionEntity(): ReplacementPortionEntity {
    return ReplacementPortionEntity(
            id = id,
            name = name,
            product = product.toPortionProductEntity(),
            sku = sku,
            unitId = unitId,
            unitNumber = unitNumber
    )
}